### PR TITLE
feat: ignore-compatibility-db

### LINKED ISSUE #5132: PNPM mysteriously installs unrelated packages
### Code to reproduce the issue:

1. Create an empty project:
   ```
   C:\>md test
   C:\>cd test
   C:\test>pnpm init
   ```
2. Install the `@babel/parser` package:
   ```
   C:\test>pnpm install @babel/parser@7.17.8

### Expected behavior:

Only `@babel/parser` should get installed, because its **package.json** file specifies no dependencies:

**@babel/parser/package.json**
```js
{
  "name": "@babel/parser",
  "version": "7.17.8",
  "description": "A JavaScript parser",
  "author": "The Babel Team (https://babel.dev/team)",
  "homepage": "https://babel.dev/docs/en/next/babel-parser",
  "bugs": "https://github.com/babel/babel/issues?utf8=%E2%9C%93&q=is%3Aissue+label%3A%22pkg%3A+parser+%28babylon%29%22+is%3Aopen",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "keywords": [
    "babel",
    "javascript",
    "parser",
    "tc39",
    "ecmascript",
    "@babel/parser"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/babel/babel.git",
    "directory": "packages/babel-parser"
  },
  "main": "./lib/index.js",
  "types": "./typings/babel-parser.d.ts",
  "files": [
    "bin",
    "lib",
    "typings"
  ],
  "engines": {
    "node": ">=6.0.0"
  },
  "devDependencies": {
    "@babel/code-frame": "^7.16.7",
    "@babel/helper-fixtures": "^7.17.0",
    "@babel/helper-validator-identifier": "^7.16.7",
    "charcodes": "^0.2.0"
  },
  "bin": "./bin/babel-parser.js"
}
```

### Actual behavior:

For some reason `@babel/types` and (its dependencies) `to-fast-properties` and `@babel/helper-validator-identifier` are also getting installed:

**pnpm-lock.yaml**
```yaml
lockfileVersion: 5.3

specifiers:
  '@babel/parser': 7.17.8

dependencies:
  '@babel/parser': 7.17.8

packages:

  /@babel/helper-validator-identifier/7.18.6:
    resolution: {integrity: sha512-MmetCkz9ej86nJQV+sFCxoGGrUbU3q02kgLciwkrt9QqEB7cP39oKEY0PakknEO0Gu20SskMRi+AYZ3b1TpN9g==}
    engines: {node: '>=6.9.0'}
    dev: false

  /@babel/parser/7.17.8:
    resolution: {integrity: sha512-BoHhDJrJXqcg+ZL16Xv39H9n+AqJ4pcDrQBGZN+wHxIysrLZ3/ECwCBUch/1zUNhnsXULcONU3Ei5Hmkfk6kiQ==}
    engines: {node: '>=6.0.0'}
    hasBin: true
    dependencies:
      '@babel/types': 7.18.9
    dev: false

  /@babel/types/7.18.9:
    resolution: {integrity: sha512-WwMLAg2MvJmt/rKEVQBBhIVffMmnilX4oe0sRe7iPOHIGsqpruFHHdrfj4O1CMMtgMtCU4oPafZjDPCRgO57Wg==}
    engines: {node: '>=6.9.0'}
    dependencies:
      '@babel/helper-validator-identifier': 7.18.6
      to-fast-properties: 2.0.0
    dev: false

  /to-fast-properties/2.0.0:
    resolution: {integrity: sha512-/OaKK0xYrs3DmxRYqL/yDc+FxFUVYhDlXMhRmv3z915w2HF1tnN1omB354j8VUGO/hbRzyD6Y3sA7v7GS/ceog==}
    engines: {node: '>=4'}
    dev: false
```

This doesn't repro with NPM 8.11.0 or Yarn 1.22.19 -- those package managers install `@babel/parser` only.

### Additional information:

 - `node -v` prints: 14.19.3
 - Windows, macOS, or Linux?: Windows and Mac

### PR DESCRIPTION
**Pull request by zkochan**
close #5132

### CODE REVIEW

File: packages/core/src/install/index.ts [hunk 1]
```diff
@@ -563,7 +566,9 @@ export function createReadPackageHook (
   if (!isEmpty(overrides ?? {})) {
     hooks.push(createVersionsOverrider(overrides!, lockfileDir))
   }
-  hooks.push(createPackageExtender(fromPairs(compatPackageExtensions)))
+  if (!ignoreCompatibilityDb) {
+    hooks.push(createPackageExtender(fromPairs(compatPackageExtensions)))
+  }
```
Comment (octogonz) [inline packages/core/src/install/index.ts:571]: https://github.com/pnpm/pnpm/issues/5132#issuecomment-1201571452 :

> It's also worth considering whether these fixups should be applied earlier, since `pnpmfile.cjs` is less useful if it is not the final authority about version selections.

(This part could maybe be a separate PR since it doesn't affect our use case with `ignoreCompatibilityDb=true`. But I do suspect that users would be surprised to learn that the input to `pnpmfile.cjs` does not reflect some hooks, and the actions of `pnpmfile.cjs` may get overridden by compatibility db fixups.)

**Review by octogonz — APPROVED** [review]
Thanks for the quick fix! 🙏

@zkochan Could we backport this to PNPM 6?

**Comment by zkochan** [comment]
> Could we backport this to PNPM 6?

yes
